rosidl_generator_c/test/test_invalid_initialization.c

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <limits.h>
+#include <float.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "rosidl_generator_c/string_functions.h"
+#include "rosidl_generator_c/msg/primitives.h"
+
+#define EXPECT_EQ(arg1, arg2) if (arg1 != arg2) return 1
+#define EXPECT_NE(arg1, arg2) if (arg1 == arg2) return 1
```

Comment:
Please put `arg1` as well as `arg2 in parentheses (e.g. `if ((arg1) != (arg2)) return 1` since otherwise the logic might be wrong when passing non trivial arguments.